tests/compile-fail/cast_fn_ptr_unsafe.rs

```diff
@@ -0,0 +1,10 @@
+// just making sure that fn -> unsafe fn casts are handled by rustc so miri doesn't have to
```

Comment:
That's fine, I just wanted to make sure it's not possible to cause a mir safe to unsafe cast.